support/metrics/metricsTracker.go

```diff
@@ -228,34 +276,50 @@ func (mt *MetricsTracker) SendUpdateEvent(now time.Time, success bool, millisFor
 
 // SendDeleteEvent sends the delete Amplitude event.
 func (mt *MetricsTracker) SendDeleteEvent(exit bool) error {
-	commonProps := mt.props
-	commonProps.SecondsSinceStart = time.Now().Sub(mt.botStartTime).Seconds()
 	deleteProps := deleteProps{
-		commonProps: commonProps,
-		Exit:        exit,
-		StackTrace:  string(debug.Stack()),
+		Exit:       exit,
+		StackTrace: string(debug.Stack()),
 	}
 
-	return mt.sendEvent(deleteEventName, deleteProps)
+	return mt.SendEvent(deleteEventName, deleteProps, time.Now())
 }
 
-func (mt *MetricsTracker) sendEvent(eventType string, eventProps interface{}) error {
+// SendEvent sends an event with its type and properties to Amplitude.
+func (mt *MetricsTracker) SendEvent(eventType string, eventPropsInterface interface{}, now time.Time) error {
 	if mt.apiKey == "" || mt.userID == "-1" || mt.isDisabled {
 		log.Printf("metric - not sending event metric of type '%s' because metrics are disabled", eventType)
 		return nil
 	}
 
+	trackerProps := mt.props
+	trackerProps[secondsSinceStartKey] = now.Sub(mt.botStartTime).Seconds()
+
+	cliVersion, ok := trackerProps[cliVersionKey].(string)
+	if !ok {
+		return fmt.Errorf("could not cast cli version back to string")
```

Comment:
it would be better to save the cliVersion on the `mt` directly rather than doing this check every time. they both achieve the same thing but it's cleaner this way and allows us to retain the type information about the cliVersion field (which is now lost since props is a map[string]interface{})